pkg/operator/controller/ingress/controller.go

```diff
@@ -1205,3 +1209,30 @@ func (r *reconciler) allRouterPodsDeleted(ingress *operatorv1.IngressController)
 
 	return true, nil
 }
+
+// updateInfraIngressLoadBalancerIPs updates GCP's PlatformStatus with Ingress LB IPs when the DNSType is `ClusterHosted`.
+func (r *reconciler) updateInfraIngressLoadBalancerIPs(service *corev1.Service, platformStatus *configv1.PlatformStatus) error {
+	if platformStatus != nil && platformStatus.Type == configv1.GCPPlatformType {
+		switch platformStatus.GCP.CloudLoadBalancerConfig.DNSType {
```

Comment:
I think it's safe to assume that `platformStatus.GCP` is not nil, but don't we need a nil check for `platformStatus.GCP.CloudLoadBalancerConfig`?